pandas/core/arrays/_mixins.py

```diff
@@ -185,6 +187,17 @@ def __setitem__(self, key, value):
     def _validate_setitem_value(self, value):
         return value
 
+    @overload
+    def __getitem__(self, key: int | np.integer) -> Any:
+        ...
+
+    @overload
+    def __getitem__(
+        self: NDArrayBackedExtensionArrayT,
+        key: slice | np.ndarray | list[int] | PositionalIndexerTuple,
```

Comment:
isn't this PostionalIndexer2D

Reply:
First, there is no `PositionScalar`, so I'm not sure what you are referring to.

For the `NDArrayBackedExtensionArray`, which extends `ExtensionArray`, `__getitem__()` needs to handle the `__getitem__()` arguments for `ExtensionArray`, which is `PositionalIndexer` , and then `NDArrayBackedExtensionArray` can widen the accepted arguments.   We then have

```python
PositionalIndexer = Union[int, np.integer, slice, List[int], np.ndarray]
PositionalIndexerTuple = Tuple[PositionalIndexer, PositionalIndexer]
PositionalIndexer2D = Union[PositionalIndexer, PositionalIndexerTuple]
```

For `NDArrayBackedExtensionArray.__getitem__()`, it accepts `PositionalIndexer2D`.  The overloads then indicate what are the return types for the various subtypes of `PositionalIndexer2D`.  This has separate return types if the arguments are `int | np.integer` versus `slice | np.ndarray | list[int] | PositionalIndexerTuple` .

Reply:
oh i had edited to `PositionalIndexer2D` which looks like the correct return type here

Reply:
> oh i had edited to `PositionalIndexer2D` which looks like the correct return type here

You mean "argument type", not "return type" ??

Here's the full code:
```python
    @overload
    def __getitem__(self, key: int | np.integer) -> Any:
        ...

    @overload
    def __getitem__(
        self: NDArrayBackedExtensionArrayT,
        key: slice | np.ndarray | list[int] | PositionalIndexerTuple,
    ) -> NDArrayBackedExtensionArrayT:
        ...

    def __getitem__(
        self: NDArrayBackedExtensionArrayT,
        key: PositionalIndexer2D,
    ) -> NDArrayBackedExtensionArrayT | Any:
```
The way it works is as follows with respect to the type of `key`:

- if the type is `int | np.integer`, return type is `Any`
- if the type is `slice | np.ndarray | list[int] | PositionalIndexerTuple`, return type is definitely `NDArrayBackedExtensionArrayT`
- If the type is `PositionalIndexer`, return type is either `NDArrayBackedExtensionArrayT | Any`

This makes some downstream typing work correctly by creating the separation this way.